### DISCUSSION TITLE
Upgrading to next Symfony version automatically?

### DISCUSSION THREAD

**Discussion by freiondrej-lmc** [symfony/symfony#44394] [Q&A] [answered]
Hello,
is there a way to upgrade e.g. from 5.3 to 5.4 or from 5.4 to 6.0 automatically, e.g. using automated refactoring tools like Rector? Has someone in the SF community considered to make it an official part of the release, or would that be too complicated?
🙂 

EDIT: I just found https://github.com/rectorphp/rector-symfony/ and https://github.com/rectorphp/rector-symfony/blob/main/config/sets/symfony/symfony54.php, but currently it helps only with validator constraint annotations.

**Comment by stof** [comment, marked as answer]
Well, if you have other things for which you can automate the migration, I'm quite sure https://github.com/rectorphp/rector-symfony/ will accept contributions.

note that for the migration to native type, Symfony provides a dedicated tool to migrate (see https://github.com/symfony/symfony/issues/43021)

**Reply by freiondrej-lmc** [reply]
@stof oh great! thank you for pointing me to the tool :)
